### PR TITLE
chore(Divider): update tests to new standards

### LINKED ISSUE #9684: Divider - update unit tests

### PR DESCRIPTION
**Pull request by mfrances17**
**What**: Closes #9684

### CODE REVIEW

File: packages/react-core/src/components/Divider/__tests__/Divider.test.tsx [hunk 1]
```diff
@@ -1,34 +1,98 @@
...
+
+test('Test 2xl orientation', () => {
+  render(<Divider orientation={{ '2xl': 'horizontal' }} />);
+  expect(screen.getByRole('separator')).toHaveClass('pf-m-horizontal-on-2xl');
+});
+
+test('Test default inset', () => {
+  render(<Divider inset={{ default: 'insetNone' }} />);
+  expect(screen.getByRole('separator')).toHaveClass('pf-m-inset-none');
+});
```
Comment (thatblindgeye) [inline packages/react-core/src/components/Divider/__tests__/Divider.test.tsx:63]: For these inset tests, we should try to test that the applicable class is applied for each combination of each key and value. So we'd want to make sure `pf-m-inset-sm-on-sm` applies and also `pf-m-inset-3xl-on-sm` applies, etc.

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
Some comments below. Additionally can we add tests for:

- Making sure props are spread correctly
- checking that the `role` attribute is not applied by default, but is applied when `component` !== hr

**Comment by mfrances17** [comment]
Made all suggested changes... also added some code that will verify all of the possible insets and consolidated it into one test. @thatblindgeye please take a look when you get a chance.

File: packages/react-core/src/components/Divider/__tests__/Divider.test.tsx (hunk 1 above)
Comment (mfrances17) [reply packages/react-core/src/components/Divider/__tests__/Divider.test.tsx:54]: i added some code that should test all of the possible insets, should do the trick

File: packages/react-core/src/components/Divider/__tests__/Divider.test.tsx [hunk 2]
```diff
@@ -1,34 +1,98 @@
...
+  expect(screen.getByRole('separator')).toContainHTML('<hr class="pf-v5-c-divider" />');
 });
 
-test('divider using div', () => {
-  const { asFragment } = render(<Divider component="div" />);
-  expect(asFragment()).toMatchSnapshot();
+test('Renders with vertical divider', () => {
+  render(<Divider orientation={{ default: 'vertical' }} />);
+  expect(screen.getByRole('separator')).toHaveClass('pf-m-vertical');
+});
```
[the remark this answers is not in the document]
Comment (mfrances17) [reply packages/react-core/src/components/Divider/__tests__/Divider.test.tsx]: done, thanks

File: packages/react-core/src/components/Divider/__tests__/Divider.test.tsx [hunk 3]
```diff
@@ -1,34 +1,98 @@
...
+
+test('Renders with li', () => {
+  render(<Divider component="li" />);
+  expect(screen.getByRole('separator')).toContainHTML('<li class="pf-v5-c-divider" role="separator" />');
+});
+
+test('Renders with div', () => {
+  render(<Divider component="div" />);
+  expect(screen.getByRole('separator')).toContainHTML('<div class="pf-v5-c-divider" role="separator" />');
 });
```
[the remark this answers is not in the document]
Comment (mfrances17) [reply packages/react-core/src/components/Divider/__tests__/Divider.test.tsx]: changed it accordingly, ty

File: packages/react-core/src/components/Divider/__tests__/Divider.test.tsx [hunk 4]
```diff
@@ -1,34 +1,98 @@
...
-  expect(asFragment()).toMatchSnapshot();
+test('Renders with only the class pf-v5-c-divider by default', () => {
+  render(<Divider data-testid="divider" />);
+  expect(screen.getByTestId('divider')).toHaveClass('pf-v5-c-divider', { exact: true });
 });
 
-test('divider using li', () => {
-  const { asFragment } = render(<Divider component="li" />);
-  expect(asFragment()).toMatchSnapshot();
+test('Renders with horizontal rule by default', () => {
```
[the remark this answers is not in the document]
Comment (mfrances17) [reply packages/react-core/src/components/Divider/__tests__/Divider.test.tsx:11]: removed, thanks

File: packages/react-core/src/components/Divider/__tests__/Divider.test.tsx [hunk 5]
```diff
@@ -1,34 +1,98 @@
...
-import { render } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
+import { Divider } from '../Divider';
 
-test('divider using hr', () => {
-  const { asFragment } = render(<Divider />);
-  expect(asFragment()).toMatchSnapshot();
+test('Renders with only the class pf-v5-c-divider by default', () => {
+  render(<Divider data-testid="divider" />);
+  expect(screen.getByTestId('divider')).toHaveClass('pf-v5-c-divider', { exact: true });
```
[the remark this answers is not in the document]
Comment (mfrances17) [reply packages/react-core/src/components/Divider/__tests__/Divider.test.tsx]: done - thanks!

File: packages/react-core/src/components/Divider/__tests__/Divider.test.tsx [hunk 6]
```diff
@@ -1,34 +1,109 @@
...
+  render(<Divider component="li" />);
+  expect(screen.getByRole('separator')?.tagName).toBe('LI');
+});
+
+test('Test default orientation (vertical divider)', () => {
+  render(<Divider orientation={{ default: 'vertical' }} />);
+  expect(screen.getByRole('separator')).toHaveClass('pf-m-vertical');
+});
+
+test('Test sm orientation', () => {
```
Comment (kmcfaul) [inline packages/react-core/src/components/Divider/__tests__/Divider.test.tsx:26]: These orientation tests could be combined into a single test similar to the updated insets test and checked all at once, but not a blocker. everything else lgtm

**Review by kmcfaul — APPROVED** [review]

**Review by thatblindgeye — APPROVED** [review]

**Pull request merged**